pkg/cmd/roachtest/test_impl.go

```diff
@@ -648,6 +653,22 @@ func (t *testImpl) IsBuildVersion(minVersion string) bool {
 	return t.BuildVersion().AtLeast(vers)
 }
 
+func panicHandler(_ context.Context, name string, l *logger.Logger, r interface{}) error {
+	l.Printf("panic stack trace in task %s:\n%s", name, string(debug.Stack()))
+	return fmt.Errorf("panic (stack trace above): %v", r)
```

Comment:
We probably want to match the error reporting to the existing handler in `test_runner.go`,

```
if r := recover(); r != nil && r != errTestFatal {
    // NB: we're careful to avoid t.Fatalf here, which re-panics.
    // Note that the error will be logged to a file, and the stack will
    // contain the source of the panic.
    t.Errorf("test panicked: %v", r)
}
```

Also shouldn't we check for the sentinel value (`errTestFatal`) since that's reserved for `t.Fatal`?

Reply:
> We probably want to match the error reporting to the existing handler in test_runner.go

Can do, although there's slight differences in the way panics are handled on the test goroutine level, vs. tasks. I'll update the wording on error to more closely match the above, but maybe just add the 'task' keyword in there so that we can discern that it was a task that panicked. The slight difference I'll explain below:

> Also shouldn't we check for the sentinel value (errTestFatal) since that's reserved for t.Fatal?

This is where the above logic comes into play when deciding to send a task event:
```
		case m.events <- event:
			// exit goroutine
		case <-m.ctx.Done():
			// Parent context already finished, exit goroutine.
```
`t.Fatal` cancels the context and reports an error to the test framework. The select logic ensures that the task's error does not get reported via `t.Errorf` because the context is canceled. This differs from how it's handled if a panic occurs in the main test goroutine, it requires the extra check for the sentinel value to ensure the error (panic) is not reported again as fatal has already reported it.

If we were to change the select logic above, event handling to report errors even if the context is cancelled we would need to handle the errors differently and check the panic sentinel error value etc.